### PR TITLE
Build all arroy indexes that need to be built

### PR DESCRIPTION
**Pull request by irevoire**
# Pull Request

## Related issue
Fixes https://github.com/meilisearch/meilisearch/issues/4588

## What does this PR do?
- Update arroy
- Ensure we always rebuild the arroy indexes that need to be built

### CODE REVIEW

File: milli/src/update/index_documents/mod.rs [hunk 1]
```diff
@@ -547,7 +547,7 @@ where
             pool.install(|| {
                 for k in crate::vector::arroy_db_range_for_embedder(embedder_index) {
                     let writer = arroy::Writer::new(vector_arroy, k, dimension);
-                    if writer.is_empty(wtxn)? {
+                    if !writer.need_build(wtxn)? && writer.is_empty(wtxn)? {
```
Comment (dureuill) [inline milli/src/update/index_documents/mod.rs:550]: hello, do we still need the `writer.is_empty` part? I would expect `need_build` to suffice?

Comment (irevoire) [reply milli/src/update/index_documents/mod.rs:550]: Huum I thought it was possible that you end up with a writer that doesn't need to be built even though the next one needs to be built?
To me, we are looking specifically for the last writer that doesn't need to be built AND that is not used by any document.
But I could be wrong. Let me know!

> it sounds strange to me to have no writer doesn't need build and then do it anyway...

```
                    if writer.need_build(wtxn)? {
                        writer.build(...)?;
                    } else if writer.is_empty(wtxn)? {
                        break;
                    }
```

Comment (dureuill) [reply milli/src/update/index_documents/mod.rs:550]: alright good catch. The revised version you suggest expresses the intention better

**Comment by dureuill** [comment]
Hello again,

tests don't pass

**Comment by irevoire** [comment]
We should be good now

**Review by dureuill — APPROVED** [review]
bors merge

**Pull request merged**